WordPress/src/main/java/org/wordpress/android/ui/domains/DomainRegistrationDetailsFragment.kt

```diff
@@ -86,7 +86,7 @@ class DomainRegistrationDetailsFragment : Fragment() {
 
         mainViewModel = ViewModelProviders.of(activity!!, viewModelFactory)
                 .get(DomainRegistrationMainViewModel::class.java)
-        viewModel = ViewModelProviders.of(this, viewModelFactory)
+        viewModel = ViewModelProviders.of(activity!!, viewModelFactory)
```

Comment:
Can you please elaborate why we need to make this change. We might want to consider adding a comment into the code as it's not a usual practice.

Note:
I feel like this viewModel should be tied to the lifecycle of the fragment and not the lifecycle of the activity. It shouldn't be an issue in most cases, but could be. We could consider replacing `activity` with `targetFragment` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/21f1484cf40df0f8129a991dcf586f7d7188b0de/WordPress/src/main/java/org/wordpress/android/ui/domains/DomainRegistrationDetailsFragment.kt#L480) and [here](https://github.com/wordpress-mobile/WordPress-Android/blob/21f1484cf40df0f8129a991dcf586f7d7188b0de/WordPress/src/main/java/org/wordpress/android/ui/domains/DomainRegistrationDetailsFragment.kt#L434). Would that work?
Wdyt?

Reply:
Yes, the reason you mentioned is the one :)

I replaced `activity` with `targetFragment`.